### PR TITLE
Make dbsync ignore last event and add timestamp to FIM events

### PR DESCRIPTION
**Pull request by JoseAntonioMHerrera**
|Related issue|
|---|
|#12166|

## Description

This PR aims to add the last_event field in fim events and make dbsync ignore it while processing the fim_entry.

## Tests

- Compilation without warnings in every supported platform
  - [X] Linux
  - [X] Windows
  - [ ] MAC OS X
- [X] Source installation

- Memory tests for Linux
  - [X] Scan-build report
- Memory tests for Windows
  - [X] Scan-build report

### CODE REVIEW

**Review by FrancoRivero — CHANGES_REQUESTED** [review]
Review Syscheckd RTR because this tests is failing

File: src/syscheckd/src/create_db.c [hunk 1]
```diff
@@ -286,7 +287,12 @@ static void transaction_callback(ReturnTypeCallback resultType, const cJSON* res
     cJSON_AddNumberToObject(data, "version", 2.0);
     cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[txn_context->evt_data->mode]);
     cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[txn_context->evt_data->type]);
-    //cJSON_AddNumberToObject(data, "timestamp", time(NULL)); This should be the last_event field
+
+    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL){
+        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
+    }else{
+        cJSON_AddNumberToObject(data, "timestamp", txn_context->latest_entry->file_entry.data->last_event);
+    }
```
Comment (antoniomanuelfr) [inline src/syscheckd/src/create_db.c:295]:
```suggestion
    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL) {
        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
    } else {
        cJSON_AddNumberToObject(data, "timestamp", txn_context->latest_entry->file_entry.data->last_event);
    }
```

File: src/syscheckd/src/registry/registry.c [hunk 2]
```diff
@@ -160,6 +161,11 @@ static void registry_key_transaction_callback(ReturnTypeCallback resultType,
     cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[event_data->evt_data->mode]);
     cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[event_data->evt_data->type]);
     cJSON_AddStringToObject(data, "arch", arch == ARCH_32BIT ? "[x32]" : "[x64]");
+    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL){
+        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
+    }else{
+        cJSON_AddNumberToObject(data, "timestamp", key->last_event);
+    }
```
Comment (antoniomanuelfr) [inline src/syscheckd/src/registry/registry.c:168]:
```suggestion
    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL) {
        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
    } else {
        cJSON_AddNumberToObject(data, "timestamp", key->last_event);
    }
```

File: src/syscheckd/src/registry/registry.c [hunk 3]
```diff
@@ -293,6 +300,13 @@ static void registry_value_transaction_callback(ReturnTypeCallback resultType,
     cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[event_data->evt_data->type]);
     cJSON_AddStringToObject(data, "arch", arch == ARCH_32BIT ? "[x32]" : "[x64]");
     cJSON_AddStringToObject(data, "value_name", name);
+
+    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL){
+        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
+    }else{
+        cJSON_AddNumberToObject(data, "timestamp", value->last_event);
+    }
+
```
Comment (antoniomanuelfr) [inline src/syscheckd/src/registry/registry.c:309]:
```suggestion

    if(timestamp = cJSON_GetObjectItem(dbsync_event, "last_event"), timestamp != NULL) {
        cJSON_AddNumberToObject(data, "timestamp", timestamp->valueint);
    } else {
        cJSON_AddNumberToObject(data, "timestamp", value->last_event);
    }

```

**Review by antoniomanuelfr — CHANGES_REQUESTED** [review]
Just some style fixes.

Also, update the base branch to the latest chamges

File: src/syscheckd/src/registry/registry.c (hunk 3 above)
Comment (JoseAntonioMHerrera) [reply src/syscheckd/src/registry/registry.c]: Done in [commit](https://github.com/wazuh/wazuh/pull/12359/commits/224cd150ea90378c265d90b500fd7e98d723ab46)

File: src/syscheckd/src/create_db.c (hunk 1 above)
Comment (JoseAntonioMHerrera) [reply src/syscheckd/src/create_db.c]: Done in [commit](https://github.com/wazuh/wazuh/pull/12359/commits/224cd150ea90378c265d90b500fd7e98d723ab46)

File: src/syscheckd/src/registry/registry.c (hunk 2 above)
Comment (JoseAntonioMHerrera) [reply src/syscheckd/src/registry/registry.c]: Done in [commit](https://github.com/wazuh/wazuh/pull/12359/commits/224cd150ea90378c265d90b500fd7e98d723ab46)

**Review by FrancoRivero — CHANGES_REQUESTED** [review]
Review tests for syscheck and create a branch inside wazuh-packages in order to see if manager side is working fine

**Review by chemamartinez — APPROVED** [review]

**Pull request merged**